openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/RequestBodyGenerator.java

```diff
@@ -83,32 +75,30 @@ public RequestBodyGenerator(RequestBody requestBody) {
     public RequiredParameterNode createNodeForRequestBody() throws BallerinaOpenApiException {
         // type CustomRecord record {| anydata...; |};
         // public type PayloadType string|json|xml|byte[]|CustomRecord|CustomRecord[] ;
-        List<Node> literals = new ArrayList<>();
-        MappingConstructorExpressionNode annotValue = null;
         Optional<TypeDescriptorNode> typeName;
         // Filter same data type
-        HashSet<Map.Entry<String, MediaType>> equalDataType = filterMediaTypes(requestBody);
-        if (!equalDataType.isEmpty()) {
-            typeName = getNodeForPayloadType(equalDataType.iterator().next());
-            SeparatedNodeList<MappingFieldNode> fields = fillRequestAnnotationValues(literals, equalDataType);
-            annotValue = NodeFactory.createMappingConstructorExpressionNode(
-                    createToken(SyntaxKind.OPEN_BRACE_TOKEN), fields, createToken(SyntaxKind.CLOSE_BRACE_TOKEN));
-        } else {
-            Iterator<Map.Entry<String, MediaType>> content = requestBody.getContent().entrySet().iterator();
-            Map.Entry<String, MediaType> mime = content.next();
-            equalDataType.add(mime);
+        List<String> types = new ArrayList<>();
+        for (Map.Entry<String, MediaType> mime : requestBody.getContent().entrySet()) {
             typeName = getNodeForPayloadType(mime);
-            if (mime.getKey().equals(GeneratorConstants.APPLICATION_URL_ENCODE)) {
-                SeparatedNodeList<MappingFieldNode> fields = fillRequestAnnotationValues(literals, equalDataType);
-                annotValue = NodeFactory.createMappingConstructorExpressionNode(
-                        createToken(SyntaxKind.OPEN_BRACE_TOKEN), fields, createToken(SyntaxKind.CLOSE_BRACE_TOKEN));
+            if (typeName.isPresent()) {
+                types.add(typeName.get().toString());
+            } else {
+                types.add(HTTP_REQUEST);
             }
         }
-        AnnotationNode annotationNode = getAnnotationNode(GeneratorConstants.PAYLOAD_KEYWORD, annotValue);
+        if (types.size() > 1 && !types.contains(HTTP_REQUEST)) {
+            String result = String.join(PIPE, types);
+            typeName = Optional.of(NodeParser.parseTypeDescriptor(result));
+        } else if (types.size() > 1 && types.contains(HTTP_REQUEST)) {
+            typeName = Optional.of(NodeParser.parseTypeDescriptor(HTTP_REQUEST));
```

Comment:
Lets simplify the logic in here as follows
```suggestion
        if (types.size() > 1 && types.contains(HTTP_REQUEST)) {
             typeName = Optional.of(NodeParser.parseTypeDescriptor(HTTP_REQUEST));
        } else if (types.size() > 1) {
             String result = String.join(PIPE, types);
            typeName = Optional.of(NodeParser.parseTypeDescriptor(result));
```